puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/foreman-proxy-develop-test.groovy

```diff
@@ -0,0 +1,89 @@
+def commit_hash = ''
+
+pipeline {
+    agent any
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage("Collect Git Hash") {
+            steps {
+                git(url: 'https://github.com/theforeman/smart-proxy', branch: 'develop')
+                script {
+                    commit_hash = archive_git_hash()
+                }
+            }
+        }
+        stage("test ruby-2.1 & puppet-4.10.9") {
```

Comment:
Previously all testing was done in parallel, possibly on different slaves. Is it possible to do the same in pipelines? Otherwise the tests will take a lot longer.

Reply:
`foreman-proxy-develop-test` took about 16 minutes to run when I tried it just now. That time includes waiting for `foreman-proxy-develop-release` to finish.

I don't consider this a significant negative.

Reply:
@ehelms suggested that if you use `agent` you can actually run them on separate nodes and in parallel but I guess we can evaluate this at a later stage.